test/common/stats/stats_matcher_test.cc

```diff
@@ -0,0 +1,246 @@
+#include "envoy/config/metrics/v2/stats.pb.h"
+
+#include "common/stats/stats_matcher_impl.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+using testing::IsFalse;
+using testing::IsTrue;
+
+namespace Envoy {
+namespace Stats {
+
+class StatsMatcherTest : public testing::Test {
+public:
+  StatsMatcherTest() {}
+  envoy::type::matcher::StringMatcher* inclusionList() {
+    return stats_config_.mutable_stats_matcher()->mutable_inclusion_list()->add_patterns();
+  }
+  envoy::type::matcher::StringMatcher* exclusionList() {
+    return stats_config_.mutable_stats_matcher()->mutable_exclusion_list()->add_patterns();
+  }
+  void rejectAll(const bool should_reject) {
+    stats_config_.mutable_stats_matcher()->set_reject_all(should_reject);
+  }
+
+  void initMatcher() { stats_matcher_impl_ = std::make_unique<StatsMatcherImpl>(stats_config_); }
+
+  envoy::config::metrics::v2::StatsConfig stats_config_;
+
+protected:
+  void expectAccepted(std::vector<std::string> expected_to_pass) {
+    for (const auto& stat_name : expected_to_pass) {
+      EXPECT_THAT(stats_matcher_impl_->rejects(stat_name), IsFalse());
```

Comment:
Usually we'd use
```
EXPECT_FALSE(stats_macher_impl_->rejects(stat_name)) << stat_name;
```
any particular reason to use the more complex form?

Reply:
No reason, will update.